app/controllers/meetings_controller.rb

```diff
@@ -3,32 +3,31 @@
 class MeetingsController < ApplicationController
   before_action :set_meeting, only: %i[show edit update destroy]
 
-  # GET /meetings
-  # GET /meetings.json
   def index
-    @meetings = Meeting.all
+    @resource = Resource.find(params[:resource_id])
+    authorize @resource, :show_meetings?
+    @meetings = policy_scope(Meeting).includes(:resource).where(resource: @resource).order(:start_time)
```

Comment:
I'm not sure I fully understand this part. Is a Policy Scope, just calling [the `resolve` method](https://github.com/firstdraft/checkins/pull/91/files#diff-0530df7cbf0b95d5caad2b245e02c839R37-R39) from the Scope class in the `MeetingPolicy`? I don't really see how that authorizes anything.

Reply:
You're correct. Calling `policy_scope` doesn't handle any of the authorization but you're supposed to use it to explicitly control in the policy file which records to show.

The authorization for this particular action is unique from others in the app because of the nested resources. It'll be helpful to review how Pundit works in a typical use case. 

In its standard usage, when you call the `authorize` method, you pass it the record you want to authorize. Based on the record's class, it finds the appropriate policy file and based on the controller action in which the `authorize` method was called, it runs the corresponding method. 

**Example:**
- I call `authorize @meeting` in `Meetings#show`. 
- Because `@meeting` is a `Meeting`, the `MeetingPolicy` class/file is found
- Because I called `authorize` in `Meetings#show`, the `show?` method in `MeetingPolicy` is run
- `show?` results in a boolean value
  - `true` simply lets the controller logic proceed
  - `false` throws an error which gets rescued in `ApplicationController` by `rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized`. 

So, getting back to the peculiar usage in `Meetings#index`. Let's say we're navigating to `/resources/26/meetings`. We want to authorize whether the current user is authorized to access the meetings specifically for Resource 26. If you pass `@meetings` to `authorize` it's going to use `MeetingPolicy` and it will not have passed in the resource (26). So instead we pass `@resource` to `authorize` which uses `ResourcePolicy` and pass an option to use a custom method instead of `index?` since you don't necessarily want the same permissions for `Meetings#index` as `Resource#index`. The resulting authorization call is like so: 
```
authorize @resource, :show_meetings?
```
and `show_meetings?` is defined as such:
```
def show_meetings?
  enrollment.teacher? && enrollment.in?(resource.enrollments)
end
```

So now, if the current user is a teacher and they are enrolled in the resource they're trying to access, they can view it!

Sorry for the very long, very granular explanation.

Reply:
@LoganDSPrice I appreciate the write up for this question! I'm not sure I understand how

this:
```ruby
authorize @resource, :show_meetings?
@meetings = policy_scope(Meeting).includes(:resource).where(resource: @resource).order(:start_time)
```
functions differently than this:
```ruby
authorize @resource, :show_meetings?
@meetings = Meeting.includes(:resource).where(resource: @resource).order(:start_time)
``` 

Since at this point, you would have already authorized the User based on the Resource, and proved that the User is a teacher, so they have access to see all the meetings. To me, it looks like both of these queries will produce the same list of `Meeting`s. Is it just better practice to use a Policy Scope with Pundit? Or am I missing something?

Reply:
I think the idea with Pundit is that in every action you use either `authorize` or `policy_scope` (or maybe both if necessary?). [This](https://github.com/varvet/pundit#ensuring-policies-and-scopes-are-used) is a good reference for how that's maintained in development. 

I think the idea for an index action in a controller that requires authorization is that you might not want authorization but you'll probably want to set up what records to return based on the user and they think that may as well live in the policy class as it is related to authorization.

Reply:
Got it! Thanks for the explanation.